main.py

```diff
@@ -971,6 +1019,15 @@ def suggest(limit=10):
     return better_jsonify(suggestions=suggestions)
 
 
+@app.route("/locations", methods=["GET"])
+@max_age(seconds=5 * 60)
+def locations():
+    kind = request.args.get("kind")
+    locs = top_locations(limit=100, kind=kind)
```

Comment:
Going forward, 100 locations is probably too much - include as a constant at the top of the source file?